### ISSUE TITLE
Disabling litellm logs

### ISSUE THREAD

**Issue by NumberChiffre** [stanfordnlp/dspy#1957]
Hey guys,

Since using 2.6.0rc3 and jumping from 2.5.42, I noticed the litellm logs are not disabled by default. I tried:
```python
dspy.disable_litellm_logging()
```

Which still shows the logs:

```shell
20:27:26 - LiteLLM:INFO: utils.py:2678 - 
LiteLLM completion() model= gpt-4o-mini; provider = openai
2024-12-18 20:27:26 [INFO] LiteLLM: 
LiteLLM completion() model= gpt-4o-mini; provider = openai
20:27:27 - LiteLLM:INFO: utils.py:921 - Wrapper: Completed Call, calling success_handler
```

How do we fully disable any litellm logs?

**Comment by okhat** [comment]
Thank you @NumberChiffre ! Are you saying this was OK in 2.5.42? Maybe it's because LiteLLM versions changed, since I can't recall that we changed this in DSPy.

**Comment by NumberChiffre** [comment]
> Thank you @NumberChiffre ! Are you saying this was OK in 2.5.42? Maybe it's because LiteLLM versions changed, since I can't recall that we changed this in DSPy.

Yeah, I stayed between 2.5.39 and 2.5.42 for a few weeks before upgrading yesterday!

**Comment by arnavsinghvi11** [comment]
Hey @NumberChiffre , 

if you do `litellm._logging._disable_debugging()` before your code, does this resolve the issue? If so, I can update the `dspy.disable_litellm_logging()` function to do this as well. 

Seems like `litellm.suppress_debug_info` (which is what's currently done by `dspy.disable_litellm_logging()`) doesn't suppress all logging for some reason

as per some of these issues -
https://github.com/BerriAI/litellm/issues/2256
https://github.com/BerriAI/litellm/issues/5403

**Comment by NumberChiffre** [comment]
@arnavsinghvi11 thanks for reply, `litellm._logging._disable_debugging()` didn't work, but this solution worked:
https://github.com/BerriAI/litellm/issues/2256#issuecomment-2041374430

@okhat Maybe this can be used as a solution on DSPy side.

**Comment by laphang** [comment]
I'm getting this as well on 2.5.43

**Comment by laphang** [comment]
@NumberChiffre that solution worked for me too (and even with logging.WARNING instead of logging.CRITICAL + 1)

**Comment by Sciumo** [comment]
Can we just remove the hard dependency on LiteLLM?
I have no end of build issues with it, and it's just a bridge pattern that can be externalized.
Actually, I'll do that in my fork.

**Comment by allenanie** [comment]
I encountered this issue as well -- copying and pasting the linked solution here:

```
loggers = [
"LiteLLM Proxy",
"LiteLLM Router",
"LiteLLM",

#added these packages ones below
"openai",
"httpx",
"urllib3.connectionpool",
"dspy.clients",
"google.auth.transport.requests",
"google.auth",
"boto3",
"botocore",
"aiobotocore"
]
for logger_name in loggers:
logger = logging.getLogger(logger_name)
logger.setLevel(logging.WARNING)
```

LiteLLM's info dump was driving me crazy.
